arangod/IResearch/IResearchFeature.cpp

```diff
@@ -163,6 +165,42 @@ void registerRecoveryHelper() {
   }
 }
 
+void transactionStateRegistrationCallback(
+    TRI_voc_cid_t cid,
+    arangodb::TransactionState& state
+) {
+  auto* vocbase = state.vocbase();
+
+  if (!vocbase) {
+    LOG_TOPIC(WARN, arangodb::iresearch::IResearchFeature::IRESEARCH)
+      << "failure to find vocbase while processing a TransactionState by IResearchFeature for tid '" << state.id() << "' cid '" << cid << "'";
+
+    return;
+  }
+
+  auto view = vocbase->lookupView(cid);
```

Comment:
It seems that this is now done at least twice per transaction.
For example, if I run `db.<collection>.insert({})`, I see two invocations of transactionStateRegistrationCallback, and two invocations of lookupView. This is definitely something we want to avoid, because it may make single-document operations more expensive.